clang/tools/driver/cc1depscand_main.cpp

```diff
@@ -617,9 +617,10 @@ int cc1depscand_main(ArrayRef<const char *> Argv, const char *Argv0,
         auto DiagsConsumer = std::make_unique<TextDiagnosticPrinter>(
             DiagsOS, new DiagnosticOptions(), false);
 
+        auto ProxyFS = FS->createProxyFS();
         SmallVector<const char *> NewArgs;
         auto RootID = updateCC1Args(
-                          *FS, *Tool, *DiagsConsumer, Argv0, Args,
+                          *ProxyFS, *Tool, *DiagsConsumer, Argv0, Args,
                           WorkingDirectory, NewArgs, PrefixMapping,
                           [&](const Twine &T) { return Saver.save(T).data(); });
```

Comment:
I think `updateCC1Args` should be in charge of this logic, since it's also the one (unsafely) calling `setCurrentWorkingDIrectoyr()`.

Moreover, in `updateCC1Args` you should be able to grab the already-constructed ProxyFS out of the DependencyScanningTool.

Reply:
(you'll need to add an accessor API to get the CachedFS out)